### PR TITLE
Code Improvements

### PR DESCRIPTION
**Pull request by bdlukaa**
Several code improvements all around the project. This implements the following things:

- Accessibility support for all the widgets
- Diagnostics support for almost all the widgets and classes (devtools)
- `debugCheckHasMacosTheme` now has a better error message
- Improved documentation all around the package
- Replaced `todo` -> `TODO`, because of dart analyzer
- Replaced `CupertinoDynamicColor` -> `DynamicColorX`
- Added `MacosTheme.maybeOf`, which can result in a nullable `MacosThemeData`

## Pre-launch Checklist

- [x] I have ran `dartfmt` on all changed files
- [x] I have ran "optimize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: lib/macos_ui.dart [hunk 1]
```diff
@@ -14,7 +12,6 @@ export 'package:flutter/material.dart'
         FlutterLogo,
         CircleAvatar;
 
-/// todo: package-level docs
```
Comment (GroovinChip) [inline lib/macos_ui.dart:17]: Let's leave this in, because we do need package-level docs

File: lib/macos_ui.dart [hunk 2]
```diff
@@ -1,7 +1,5 @@
 library macos_ui;
 
-export 'package:flutter/cupertino.dart'
```
Comment (GroovinChip) [inline lib/macos_ui.dart:3]: This removal reminds me that we really shouldn't be exporting any non-macos_ui libraries in this file, it messes with the generation of documentation. If we need the convenience of exporting flutter libraries in one file, we should have a separate file specifically for this that is not exported in `lib/macos_ui.dart`. Perhaps a `lib/src/library.dart`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
+## [0.0.9+1]
+* `CapacityIndicator` colors can now be setted on its constructor
```
Comment (GroovinChip) [inline CHANGELOG.md]: "setted" isn't a word, change to something like "`CapacityIndicator` colors can now be set on its constructor"

**Review by GroovinChip — CHANGES_REQUESTED** [review]
These changes are great, I especially like the accessibility improvements!

File: lib/macos_ui.dart (hunk 2 above)
Comment (GroovinChip) [reply lib/macos_ui.dart:3]: I will tinker with this now

File: CHANGELOG.md (hunk 3 above)
Comment (bdlukaa) [reply CHANGELOG.md]: Oh sorry. I didn't know that

Comment (GroovinChip) [reply CHANGELOG.md]: No worries 😊

File: lib/macos_ui.dart (hunk 2 above)
Comment (GroovinChip) [reply lib/macos_ui.dart:3]: Made the change, running dartdoc now to see the results

**Review by GroovinChip — CHANGES_REQUESTED** [review]
Looking good. I will submit my own changes on this branch shortly.

**Review by GroovinChip — APPROVED** [review]

**Comment by GroovinChip** [comment]
Other changes made:
* Updated `dart_code_metrics` to latest version
* Added a lint rule to avoid relative lib imports

**Pull request merged**